app/controllers/events_controller.rb

```diff
@@ -96,4 +120,58 @@ def date_from_form(date_info)
     def event_params
       params.require(:event).permit(:name, :description, :max_participants, :active, :kind, :organizer, :knowledge_level)
     end
+
+    # Generate all names to print from the query-params
+    #
+    # @return participant_names as array of strings
+    def badges_name_params
+      participant_names = []
+      params.each do | k, v |
+        # check if key is containing the keyword, schema of name_param: #{participant.id}_print_#{participant.name}
+        if k.include? "_print_"
+          name_index = k.index("_print_") + "_print_".chars.length
+          participant_names.push(k[name_index .. k.chars.length])
+        end
+      end
+      participant_names
+    end
+
+    # Create a name badge in a given pdf
+    #
+    # @param pdf, is a prawn pdf-object
+    # @param name [String] is the name label of the new badge
+    # @param x [Integer] is the x-coordinate of the upper left corner of the new badge
+    # @param y [Integer] is the y-coordinate of the upper left corner of the new badge
+    def create_badge(pdf, name, x, y)
+      width = 260
+      height = 150
+      pdf.stroke_rectangle [x, y], width, height
+      pdf.draw_text name, :at => [x + width / 2 - 50 , y - 20]
+    end
+
+    # Create a page with maximum 10 badges
+    #
+    # @param pdf, is a prawn pdf-object
+    # @param names [Array of Strings] are the name which are printed to the badges
+    # @param index [Number] the page number
+    def create_badge_page(pdf, names, index)
+      # create no pagebreak for first page
+      if index > 0
+        pdf.start_new_page
+      end
+      # creates badge edges as rectangles left-upper-bound[x,y], width, height
+      names.each_with_index do |participant, index|
+        if index % 2 == 0 # left column badges
```

Comment:
Instead of working with indices and integer division, you could use
```ruby
names.each_slice(2).with_index do |(left, right), row|
  create_badge(pdf, left, 0, 750 - row * 150)
  create_badge(pdf, right, 260, 750 - row * 150) unless right.nil?
end
```